### PR TITLE
add the LineRef filtering parameter

### PR DESCRIPTION
**Pull request by antoine-de**
this StopMonitoring parameter filter on some lines

### CODE REVIEW

File: src/routes/stop_monitoring.rs [hunk 1]
```diff
@@ -76,6 +78,13 @@ fn create_monitored_stop_visit(
     }
 }
 
+fn get_line_ref<'a>(cnx: &Connection, model: &'a navitia_model::Model) -> &'a str {
+    let vj = &model.vehicle_journeys[cnx.dated_vj.vj_idx];
+    let route = model.routes.get(&vj.route_id).unwrap(); // should we have this unwrap ? do we consider that we can't have bad data ?
```
Comment (antoine-de) [inline src/routes/stop_monitoring.rs]: what do you think about this @Tristramg ?

File: src/routes/stop_monitoring.rs [hunk 2]
```diff
@@ -44,6 +44,10 @@ fn create_monitored_stop_visit(
 ) -> model::MonitoredStopVisit {
     let stop = &data.ntm.stop_points[connection.stop_point_idx];
     let vj = &data.ntm.vehicle_journeys[connection.dated_vj.vj_idx];
+    let route = &data.ntm.routes.get(&vj.route_id);
+    let line_ref = route
+        .map(|r| r.line_id.clone())
+        .unwrap_or_else(|| "".to_owned());
```
Comment (TeXitoi) [inline src/routes/stop_monitoring.rs]: With get_corresponding, you can have the line(s) from the vjs directly (as a ptref request)

File: src/routes/stop_monitoring.rs (hunk 1 above)
Comment (TeXitoi) [reply src/routes/stop_monitoring.rs]: A model can't be build without I'd coherency. get_corresponding allow you to get the line from the vj without using the route explicitly

File: src/routes/stop_monitoring.rs (hunk 2 above)
Comment (antoine-de) [reply src/routes/stop_monitoring.rs]: oh great, I forgot this!

thanks I'll use it tomorrow

**Pull request merged**